### DISCUSSION TITLE
Automatisiertes Update von 1.6 auf 2.0 - freiwillige Tester gesucht!

### DISCUSSION THREAD

**Discussion by camueller** [camueller/SmartApplianceEnabler#354] [Installation und Betrieb]
Gestern und heute habe ich den ganzen Tag mit der Erstellung eines automatisierten Updates verbracht: Idee gehabt, Script erstellt oder angepasst, Update durchgeführt, Verbesserungsbedarf festgestellt und wieder von vorn ...

Wichtig war mir, dass der Update-Prozess auch via `webmin` funktioniert, da es doch einige User gibt, welche SSH nicht nutzen wollen/können.

Bevor ich eine 2.0-Version als "offizielle" Version deklariere und damit viele SAE-User zum Update veranlasse, möchte ich den Update-Prozess erstmal mit einer kleinen Anzahl von Usern (ca. 3-5) testen, welche folgende Bedingungen erfüllen sollten:
- mindestens ein Gerät nutzt GPIO (also z.B. S0-Zähler)
- in der Lage sein, die Konfigurationsdateien selbständig vor dem Update zu sichern (im Rahmen des Updates werden sie nochmal gesichert)
- sich die Dokumentation [Update von Version 1.6 auf Version 2.x](https://github.com/camueller/SmartApplianceEnabler/blob/master/doc/Update_DE.md) ansehen und die Durchführung zutraun
- sich bewusst machen, dass es keinerlei Ansprüche dafür gibt, dass alles funktioniert (das gilt generell für die Nutzung des SAE und auch für Updates zu einem späteren Zeitpunkt!)

Der SAE 2.0 kann alles, was die Version 1.6 auch kann. Darüber hinaus hat die Version 2.0 folgende Vorteile
- MQTT-basiert für bessere Integration in Hausautomatisierungen
- detaillierteres, konfigurierbares Dashboard auf Basis von NodeRED (da habe ich leider bisher wenig Feedback erhalten)
- zusätzliche Zähler- und Schalterarten
- Erkennung des verbundenden Fahrzeugs bei Konfiguration mehrerer Fahrzeuge mit SOC-Script

Ich selbst nutze den SAE 2.0 seit vielen Monaten auf meinen beiden produktiven Raspis.

Interessierte User sollten das bitte in diesem Thread kundtun, aber das Update noch nicht beginnen! Für die Durchführung des Updates sollten wir einen Zeitraum abstimmen, in dem ich dazu in der Lage bin, kurzfristig kostenfreien Support zu geben, damit der Raspi schnell wieder funktioniert, falls beim Update Probleme auftreten.

Die Dokumentation für ein manuelles Update von Version 1.6 auf 2.0 kommt noch, aber das werden im wesentilichen Verweise auf die bereits vorhandenene Dokumentation der [manuellen Installation](https://github.com/camueller/SmartApplianceEnabler/blob/2.0/doc/ManualInstallation_DE.md) sein.

**Comment by wuenschhen** [comment]


**Reply by camueller** [reply]
Ich suche User, die aktuell SAE 1.6 einsetzen - siehe Überschrift.

**Reply by stevoboy** [reply]
Ich setze 1.6.x ein.

**Reply by camueller** [reply]
Hast Du GPIO-basierte Zähler/Schalter?

**Reply by stevoboy** [reply]
> Hast Du GPIO-basierte Zähler/Schalter?

Leider nein, nur http-basiert.

**Comment by Flauschi83** [comment]
ich habe aktuell 1.6.11 installiert allerdings in proxmox als VM

**Reply by camueller** [reply]
Das automatisierte Update funktioniert nur mit Raspbian! In Deinem Falls musst Du ein manuelles Update machen, was aktuell noch nicht dokumentiert ist.

**Comment by camueller** [comment]
Es wäre schön, wenn sich noch ein paar User mit GPIO-Verwendung finden, weil sich in diesem Bereich einiges ändert beim Update von 1.6 auf 2.0 ...

**Reply by ThomasInOndrup** [reply]
Moin, ich hab zwei Raspberry Pi mit SAE 1.6.17. Einmal mit drei S0-Zählern und 4-fach-Relaisboard und einmal mit zwei S0-Zählern und einem über GPIO angesprochenen Schütz.

**Comment by camueller** [comment]
@ThomasInOndrup @wuenschhen Wenn Ihr wollt, könnt Ihr das Update in den nächsten Tagen durchführen. Bitte hier den Erfolg oder Misserfolg melden. Falls es Probleme geben sollte, kann ich kurzfristig Support geben.

**Reply by wuenschhen** [reply]
Hallo, ich könnte es am Wochenende testen. Bitte die Updateprozedure zur Verfügung stellen.  Danke.

**Reply by camueller** [reply]
Die Dokumentation ist oben im Post verlinkt in der Hoffnung, dass Ihr das durchlest, bevor ihr Euch zum Testen meldet.

**Reply by ThomasInOndrup** [reply]
Ich hab das Update nach Beschreibung über webmin durchgeführt. Webmin, obwohl ich bisher alles über SSH gemacht habe, weil ich einfach neugierig war.
Das Update auf bullseye hat deutlich über eine Stunde gedauert, aber soweit funktioniert. Auch das Update auf SAE 2.0.6 funktionierte. Ich hatte nur nicht mehr dran gedacht, dass ich ja auch noch den MQTT-Broker brauche, den ich dann nachträglich installiert und anschließend den SAE neugestartet habe. Es scheint jetzt zu funktionieren. Ich muss die Werte nochmal kontrollieren, aber auf jeden Fall bezieht der SHM jetzt Werte vom SAE.

**Reply by camueller** [reply]
Das klingt ja schon mal gut :-) 

Wieso hast Du den MQTT-Broker manuell installiert? Im Rahmen des Updates wird der MQTT-Broker Mosquitto installiert und konfiguriert.
Welchen MQTT-Broker hast Du selbst installiert? Wenn Du nur den Befehl zur Installation von Mosquitto eingegeben hast, hat der ja nichts gemacht, weil er schon installiert war.

**Reply by ThomasInOndrup** [reply]
Ich hatte nicht auf dem Schirm, dass der MQTT-Broker automatisch installiert wird.

Es funktioniert offenbar doch nicht so richtig. Mein SHM zeigt dauerhaft 0W für beide Zähler an. Über `grep S0ElectricityMeter /tmp/rolling-2022-10-19.log` bekomme ich jedoch den Verbrauch angezeigt. Woran kann das liegen?

**Reply by camueller** [reply]
Mit SHM meinst Du sicher das Portal - der SHM zeigt ja nichts an. Wird an der `/semp`-Schnittstelle die korrekte Leistungsaufnahme angezeigt? Falls ja, solltest Du mal ein Reset des SHM machen (bei SHM 2.0: Reset-Taste länger als 6s gedrückt halten).

**Reply by ThomasInOndrup** [reply]
Ja, natürlich, ich meinte das SunnyPortal. Da an der `/semp`-Schnittstelle alles richtig angezeigt wurde, habe ich den SHM kurz vom Stromnetz getrennt (hatte keine Büroklammer zur Hand). Jetzt scheint es wirklich zu funktionieren.
Kann ich denn nun auch MQTT nutzen, um die Verbraucher bei Grafana einzubinden?

**Reply by camueller** [reply]
> Ja, natürlich, ich meinte das SunnyPortal. Da an der `/semp`-Schnittstelle alles richtig angezeigt wurde, habe ich den SHM kurz vom Stromnetz getrennt (hatte keine Büroklammer zur Hand). Jetzt scheint es wirklich zu funktionieren. 

Sehr gut! Wenn das SAE-Update bei Dir mit GPIO geklappt hat (da gab es ja die "schwerwiegendste" Umstellung), solltest es auch für alle anderen funktionieren.

> Kann ich denn nun auch MQTT nutzen, um die Verbraucher bei Grafana einzubinden?

Klar, schau Dir doch einfach mal die MQTT-Nachrichten an. Ich nutze dafür den  [MQTT-Explorer](http://mqtt-explorer.com/).

**Comment by camueller** [comment]
Keine weiteren Tester mit Raspberry Pi?

**Reply by FireFly3390** [reply]
noch aktuell?
Ich hätte 1.6.21 auf Raspi 2B laufen mit einem S0 Zähler. Allerdings hab ich das Update schon einmal versucht (vor dem release) und gescheitert.

**Reply by camueller** [reply]
Inzwischen ist 2.1.0 draussen und das Updaten sollte für jeden erfolgreich möglich sein.

**Comment by sgrafcs** [comment]
Morgen,

ich nutze SAE 1.6.20 auf einem PI 3A+ mit zwei S0 Messungen und einer SGReady Steuerung,

Würde gerne auf 2.x migrieren.

**Reply by camueller** [reply]
Sehr gut! Im Eingangspost findest Du die Links zu Dokumentation. Nachdem @ThomasInOndrup, der ebenfallls GPIO verwendet, erfolgreich migriert hat, sollte es auch bei Dir keine Probleme geben.

Du schreibst, dass Du eine SGReady-Steuerung hast. Hast Du https://github.com/camueller/SmartApplianceEnabler/issues/360 verfolgt?

**Reply by sgrafcs** [reply]
Update mache ich am Wochenende.

Ja, ich hatte das mit dem SGReady verfolgt. Bei mir funktioniert das aktuell ganz leidlich allerdings habe ich manchmal das Gefühlt dass die Überschusserkennung nicht sauber ist, speziell wenn die WP selber schon Strom zieht. Mit SAE 2.x versuche ich das dann mal mit dem S0 Slave-Zähler.

SGReady hat bei mir aber eine recht geringe Prio, das es verbrauchstechnisch nicht allzu viel optimiert, da ich öfters de Auto mit Sonne lade, dann bleibt ehe nichts über.

**Reply by sgrafcs** [reply]
Soweit hat das Update funktioniert, SAE läuft und der HM2 hat auch gleich alle konfigurierten Geräte wieder erkannt und die S0-Zählöer werden auch ausgewertet.

Allerdings ist das Web-Interface von SAE tot, die Seite wird nur leer dargestellt.

In der Console kommen diese vier Meldungen:

Failed to load resource: the server responded with a status of 404 ()
:8080/vendor.ff84a140ca357345.js:1          Failed to load resource: the server responded with a status of 404 ()
:8080/main.a6f72e8def0150e9.js:1          Failed to load resource: the server responded with a status of 404 ()
styles.135aa27deb08b4f0.css:1          Failed to load resource: the server responded with a status of 404 ()

**Reply by camueller** [reply]
Lösch mal den Web-Browser-Cache für diese Seite.

**Reply by sgrafcs** [reply]
Ja, jetzt kommt die Seite, danke.

**Reply by sgrafcs** [reply]
Ich vermute fürs Web-Interface fehlen eine Resource:

<img width="356" alt="image" src="https://user-images.githubusercontent.com/32098328/198824039-24d2eae3-4af0-4b09-8903-f9645a2b6a0a.png">

<img width="305" alt="image" src="https://user-images.githubusercontent.com/32098328/198824095-b464348e-6e9c-4fe3-895a-73db0f2419c2.png">

**Reply by sgrafcs** [reply]
Ich musste den ganzen Browser-Chae löschen, dann klappt es

**Comment by tobrog** [comment]
Ich habe gerade auch erfolgreich von 1.6.21 auf 2.0.6 per ssh aktualisiert auf einem RPI4. Hab zwar nur ein paar Shelly Plug S und Gosund EP2 Steckdosen im SAE, aber die scheinen soweit zu funktionieren.

**Reply by tobrog** [reply]
Nach einem Neustart lief mosquitto nicht mehr. Fehlermeldung: "Address already in use"
Da mosquitto bei mir schon vorher auf port 1883 lief (ich hab den listener 1883 schon in /etc/mosquitto/mosquitto.conf), kam es wohl zu dem Konflikt.
Nachdem ich in /etc/mosquitto/conf.d/smartapplianceenabler.conf "listener 1883" auskommentiert habe, läuft aber wieder alles.

**Reply by camueller** [reply]
Habe das upgrade-Script geändert: Vor der Installation von `mosquitto` wird geprüft, ob er bereits installiert ist. Nur wenn das nicht der Falls ist, wird er installiert und konfiguriert.

**Comment by jozahner** [comment]
Ich bin aktuell auf 1.6.21 und würde auch gerne auf 2.x updaten, allerdings habe ich eine **manuelle Installation** (weil der Raspi noch anderweitig genutzt wird, habe ich SAE einfach dazu installiert). Könntest umschreiben was es beim manuellen Update zu beachten gibt?
MQTT so wie bei der Erstinstallation brauchts sicher, was sonst noch? *.war ersetzen und ein config backup (Appliances.xml), what else?
Auf Raspi OS bullseye bin ich schon und GPIO nutze ich nicht, ist also ne relativ simple Geschichte.

**Reply by camueller** [reply]
Die offizielle 2.1.0 kommt in den nächsten 1-2 Tagen. Darauf würde ich an Deiner Stelle noch warten.

Dokumentation zum Update: https://github.com/camueller/SmartApplianceEnabler/blob/master/doc/Update_DE.md

**Reply by jozahner** [reply]
Merci für den Tipp bezüglich der neuen Version.
Den Upgrade Guide habe ich gesehen, was mir nicht klar war, war folgender Satz:
"Das Update des Smart Appliance Enabler erfolgt, wie ganz oben auf dieser Seite beschrieben." -> Da wärs gut wenn du etwas genauer schreiben würdest was du meinst, referenzierst du dich einfach auf den Austausch vom *.war File bei gleichbeibendem Minor-Release?

**Reply by camueller** [reply]
Habe es präzisiert und einen Link eingefügt.

**Reply by jozahner** [reply]
**Upgrade** meiner **manuellen Installation** war erfolgreich. Folgendes habe ich in Kurzfassung gemacht:

1. MQTT (Mosquitto) installiert und anhand der manuellen Installationsanleitung konfiguriert
2. Appliances.xml gespeichert
3. *.war File ersetzt

Das wars, ich sehe wieder alles im SunnyPortal. Hat also geklappt. Merci auch @camueller fürs updaten des Manuals.
